src/registrar/models/user.py

```diff
@@ -64,6 +66,36 @@ def unrestrict_user(self):
     def is_restricted(self):
         return self.status == self.RESTRICTED
 
+    @classmethod
+    def needs_identity_verification(cls, email, uuid):
+        """A method used by our oidc classes to test whether a user needs email/uuid verification
+        or the full identity PII verification"""
+
+        # An existing user who is a domain manager of a domain (that is,
+        # they have an entry in UserDomainRole for their User)
+        try:
+            existing_user = cls.objects.get(username=uuid)
+            if existing_user and UserDomainRole.objects.filter(user=existing_user).exists():
+                return False
+        except cls.DoesNotExist:
+            # Do nothing when the user is not found, as we're checking for existence.
+            pass
+        except Exception as err:
+            raise err
+
+        # A new incoming user who is a domain manager for one of the domains
+        # that we inputted from Verisign (that is, their email address appears
+        # in the username field of a TransitionDomain)
+        if TransitionDomain.objects.filter(username=email).exists():
+            return False
+
+        # A new incoming user who is being invited to be a domain manager (that is,
+        # their email address is in DomainInvitation for an invitation that is not yet "retrieved").
+        if DomainInvitation.objects.filter(email=email, status=DomainInvitation.INVITED).exists():
```

Comment:
My only call out here is that we do have an issue right now where the add a User as Domain Manager button doesn't check for case, but login.gov gives us a lower case email. I don't think it matters here as that will be handled in another ticket for this sprint but felt worth mentioning since it would make this if fail (when one may think it should be enter just fine).